File: AurigainLoanERPApi/AurigainLoanERP.Shared/ContractModel/DistrictModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AurigainLoanERP.Shared.ContractModel
{
    public class DistrictModel
    {

       public  DistrictModel() 
        {
            Areas = new List<PincodeAreaModel>();
        }
        public long Id { get; set; }
        public string Name { get; set; }
        public int StateId { get; set; }
        public string Pincode { get; set; }
        public string StateName { get; set;}
        public bool? IsActive { get; set; }
        public bool IsDelete { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public List<PincodeAreaModel> Areas { get; set;}
    }
    /// <summary>
    /// District Dropdown Model
    /// </summary>
    public class DDLDistrictModel 
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Pincode { get; set;}
    }
}

File: AurigainLoanERPApp/src/app/Shared/Model/master-model/district.model.ts
export class DistrictModel {
  constructor() {
    this.Areas = [] as PincodeAreaModel[];
  }
  Id!: number;
  Name!: string | null;
  StateId!: number;
  Pincode!: string;
  StateName!: string | null;
  IsActive!: boolean | null;
  IsDelete!: boolean;
  CreatedOn!: Date;
  ModifiedOn!: Date | null;
  Areas!: PincodeAreaModel[];
}
export class DDLDistrictModel {
  Id!: number;
  Name!: string | null;
  Pincode!: string | null;
}
export class PincodeAreaModel {
  Id!: number;
  Pincode!: string;
  AreaName!: string;
  IsActive!: boolean | null;
  IsDelete!: boolean;
  DistrictId!: number;
}
